### PR TITLE
Add release GitHub action to sign Serialization DLL for release 

### PR DESCRIPTION
**Pull request by byronlin13**
*Issue #, if available:*

*Description of changes:*
This PR is to demonstrate the integration of the Ion Object mapper library and the AWS signer service(for security reviewer).

We created a GitHub action to automate the release from signing the DLL to publishing to NuGet.

The high level workflow is:
1. Build the library (which will produce unsigned DLL)
2. Push the unsigned DLL into designated unsigned S3 bucket (this will trigger the signing job)
3. Poll the signed bucket to see if the signed DLL appears
4. Fetch the signed DLL from designated signed S3 bucket

Tested successfully on my [fork](https://github.com/byronlin13/amazon-qldb-driver-dotnet/actions/runs/1031842284)

Whats remaining:
* Package the signed DLL and push to NuGet

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/release_serialization.yml [hunk 1]
```diff
@@ -32,15 +26,14 @@ jobs:
     - name: Install dependencies
       run: dotnet restore
     - name: Build
-      run: dotnet build --configuration Release --no-restore
+      run: dotnet build --configuration Release --no-restore
```
Comment (kylepbit) [inline .github/workflows/release_serialization.yml:29]: Nit: should revert whitespace change.

File: .github/workflows/release_serialization.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
+name: Serialization Library Release
+
+on: workflow_dispatch
```
Comment (kylepbit) [inline .github/workflows/release_serialization.yml:3]: Is this saying it's now a manual trigger?

Comment (simonz-bq) [reply .github/workflows/release_serialization.yml:3]: Yes

File: .github/workflows/release_driver.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        dotnet: ['5.0.x']
+      
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_SIGNING_ACCOUNT }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_SIGNING_ACCOUNT }}
```
Comment (guyilin-amazon) [inline .github/workflows/release_driver.yml:55]: Do we have the secret of `AWS_ACCESS_KEY_ID_SIGNING_ACCOUNT` and `AWS_SECRET_ACCESS_KEY_SIGNING_ACCOUNT`?

**Review by alpian — APPROVED** [review]

**Pull request merged**